urllib3/poolmanager.py

```diff
@@ -152,7 +202,11 @@ def urlopen(self, method, url, redirect=True, **kw):
         :class:`urllib3.connectionpool.ConnectionPool` can be chosen for it.
         """
         u = parse_url(url)
-        conn = self.connection_from_host(u.host, port=u.port, scheme=u.scheme)
+        request_context = self.connection_pool_kw.copy()
+        request_context['scheme'] = u.scheme or 'http'
+        request_context['port'] = u.port or port_by_scheme.get(u.scheme, 80)
+        request_context['host'] = u.host
```

Comment:
Why have you changed this to not delegate to `connection_from_host`? It seems like you've duplicated a function that was created to avoid this duplication.